clustered/common/src/main/java/org/ehcache/clustered/common/messages/ResponseCodec.java

```diff
@@ -50,20 +52,20 @@
     }
   }
 
-  public static EhcacheEntityResponse decode(byte[] payload) {
+  public EhcacheEntityResponse decode(byte[] payload) {
     ByteBuffer buffer = ByteBuffer.wrap(payload);
     byte opCode = buffer.get();
     EhcacheEntityResponse.Type type = EhcacheEntityResponse.Type.responseType(opCode);
     byte[] payArr = new byte[buffer.remaining()];
     buffer.get(payArr);
     switch (type) {
       case SUCCESS:
-        return EhcacheEntityResponse.success();
+        return EhcacheEntityResponse.Success.INSTANCE;
       case FAILURE:
-        Exception exception = (Exception)LifeCycleOpCodec.unmarshall(payArr);
-        return EhcacheEntityResponse.failure(exception);
+        Exception exception = (Exception)Util.unmarshall(payArr);
+        return new EhcacheEntityResponse.Failure(exception);
       case GET_RESPONSE:
-        return EhcacheEntityResponse.response(ChainCodec.decode(payArr));
+        return new EhcacheEntityResponse.GetResponse(ChainCodec.decode(payArr));
```

Comment:
Again another static hiding there. Now maybe this can be simply flattened in this class as I believe no one else uses the `ChainCodec`